include/deal.II/distributed/shared_tria.h

```diff
@@ -106,6 +113,39 @@ namespace parallel
                                          const std::vector< CellData< dim > > &cells,
                                          const SubCellData &subcelldata);
 
+      /**
+       * Return auxilary vector subdomain_id_with_artificial .
```

Comment:
i would prefer to put it to `private`, BUT that implies that Policy classes should be `friends` of shared::Tria. It's not clear to me how to achieve it given the fact that Policies are internal classes, not available outside. I also expect circular includes.

I can put it any closer to the place where I need it as I have to store original partition from METIS done in Triangulation class.